src/mcts/search.cc

```diff
@@ -1234,6 +1234,13 @@ void SearchWorker::ExtendNode(Node* node) {
     }
   }
 
+  // A lonely king cannot mate, so indicate that with restricted bounds.
+  if (board.ours().count() < 2) {
+    node->SetBounds(GameResult::DRAW, GameResult::WHITE_WON);
```

Comment:
How does `SetBounds` work? The best we have if only a king is left is between a loss and a draw, right?
Shouldn't this be something like (pseude code) `node->SetBounds(lower=Loss, upper=Draw)`

For some reason it looks as if SetBounds takes the values backward, but there must be a reason for this and most probably I did not yet understand how SetBounds works, how does it work?

Reply:
I don't know in this particular case for certain, but we do some things where values are stored from the parents perspective, especially around how we deal with GameResult it can be quite unintuitive at times.

Reply:
Yeah, if you look at other outcomes in `ExtendNode`:
https://github.com/LeelaChessZero/lc0/blob/9158a67a917d630caf7fbd22baeaebd31686db24/src/mcts/search.cc#L1174-L1179
https://github.com/LeelaChessZero/lc0/blob/9158a67a917d630caf7fbd22baeaebd31686db24/src/mcts/search.cc#L1221-L1227

Reply:
Yeah well... seen that before but it is pretty hard to read, I'm not entirely sure why code is like we are always talking to the parent of a leaf node (or so it seems). 

is this part responsible for that?

```
Node* cur = node;
  while (cur != search_->root_node_) {
    Node* prev = cur->GetParent();
    to_add.push_back(prev->GetEdgeToNode(cur)->GetMove());
    cur = prev;
  }
```

And to be honest I'm not sure how the code could be more readable either...